src/lazy/encoding.rs

```diff
@@ -1,39 +1,53 @@
+#![allow(non_camel_case_types)]
 use crate::lazy::binary::raw::annotations_iterator::RawBinaryAnnotationsIterator;
 use crate::lazy::binary::raw::r#struct::LazyRawBinaryStruct;
 use crate::lazy::binary::raw::reader::LazyRawBinaryReader;
 use crate::lazy::binary::raw::sequence::{LazyRawBinaryList, LazyRawBinarySExp};
 use crate::lazy::binary::raw::value::LazyRawBinaryValue;
-use crate::lazy::decoder::LazyDecoder;
-use crate::lazy::text::raw::r#struct::LazyRawTextStruct;
+use crate::lazy::decoder::{LazyDecoder, LazyMacroInvocation};
+use crate::lazy::text::raw::r#struct::LazyRawTextStruct_1_0;
 use crate::lazy::text::raw::reader::LazyRawTextReader;
-use crate::lazy::text::raw::sequence::{LazyRawTextList, LazyRawTextSExp};
-use crate::lazy::text::value::{LazyRawTextValue, RawTextAnnotationsIterator};
+use crate::lazy::text::raw::sequence::{LazyRawTextList_1_0, LazyRawTextSExp_1_0};
+use crate::lazy::text::value::{LazyRawTextValue_1_0, RawTextAnnotationsIterator};
 
 // These types derive trait implementations in order to allow types that containing them
 // to also derive trait implementations.
 
+/// Marker trait for binary encodings of any version.
+pub trait LazyBinaryDecoder {}
+
 /// The Ion 1.0 binary encoding.
 #[derive(Clone, Debug)]
-pub struct BinaryEncoding;
+pub struct BinaryEncoding_1_0;
+impl LazyBinaryDecoder for BinaryEncoding_1_0 {}
 
 /// The Ion 1.0 text encoding.
 #[derive(Clone, Debug)]
-pub struct TextEncoding;
+pub struct TextEncoding_1_0;
 
-impl<'data> LazyDecoder<'data> for BinaryEncoding {
+impl<'data> LazyDecoder<'data> for BinaryEncoding_1_0 {
     type Reader = LazyRawBinaryReader<'data>;
     type Value = LazyRawBinaryValue<'data>;
     type SExp = LazyRawBinarySExp<'data>;
     type List = LazyRawBinaryList<'data>;
     type Struct = LazyRawBinaryStruct<'data>;
     type AnnotationsIterator = RawBinaryAnnotationsIterator<'data>;
+    // Macros are not supported in Ion 1.0
+    type MacroInvocation = ();
```

Comment:
question—instead of `()`, could you use `!` here?

Reply:
I've done a version of that in the follow on PR in working on 👍